### DISCUSSION TITLE
gitPython - Pull request url. Once we do git push, is there a way to get the PR url back from the cmd line? I was not able to get any libraries return that. Is this possible?

### DISCUSSION THREAD

**Discussion by mehtapinal** [gitpython-developers/GitPython#1149] [Q&A] [answered]

**Comment by mehtapinal** [comment]
Somehow I want retrieve pull request url:

> git push origin branch_xxx
> Enumerating objects: 13, done.
> Counting objects: 100% (13/13), done.
> Delta compression using up to 12 threads
> Compressing objects: 100% (7/7), done.
> Writing objects: 100% (7/7), 754 bytes | 754.00 KiB/s, done.
> Total 7 (delta 6), reused 0 (delta 0)
> remote: Push is OK
> remote:
> remote: Create pull request for branch_xxx
**> remote:   https://bitbucket.XXX.com/projects/test/repos/monolith/pull-requests?create&sourceBranch=refs/heads/branch_xxx**

I hope i have explained myself. The only workaround I currently do is: I fabricate that URL myself within the application. But it will be good if there is a gitPython library which would do so.

**Comment by Byron** [comment, marked as answer]
I think what you can try is to implement your own [progress delegate](https://gitpython.readthedocs.io/en/stable/tutorial.html#advanced-repo-usage) (search for 'progress') to see if it gets to see the given line of the remote.

Generally GitPython sees all lines sent by the remote and definitely makes all of them available to the delegate. It's probably worth perusing [the code](https://github.com/gitpython-developers/GitPython/blob/master/git/util.py#L372:L372) to see how exactly it works, and [how it is used](https://github.com/gitpython-developers/GitPython/blob/master/git/remote.py#L659:L659).

Please feel free to post follow ups on the closed issue.
